admin/app/adapters/complementary-certification.js

```diff
@@ -6,4 +6,30 @@ export default class ComplementaryCertificationAdapter extends ApplicationAdapte
   urlForFindRecord(id) {
     return `${this.host}/${this.namespace}/complementary-certifications/${id}/target-profiles`;
   }
+
+  urlForUpdateRecord(id) {
+    return `${this.host}/${this.namespace}/complementary-certifications/${id}/badges`;
+  }
+
+  async updateRecord(store, type, complementaryCertification) {
+    if (complementaryCertification.adapterOptions?.attachBadges === true) {
+      const {detachedTargetProfileId, attachedTargetProfileId} = complementaryCertification.adapterOptions;
+      const payload = this.serialize(complementaryCertification, {includeId: true});
+      delete payload.data.attributes['target-profiles-history'];
+
+      payload.data.attributes.attachedTargetProfileId = attachedTargetProfileId;
+      payload.data.attributes.detachedTargetProfileId = detachedTargetProfileId;
+
+      const relatedBadges = complementaryCertification.hasMany('complementaryCertificationBadges') ?? [];
```

Comment:
Si cette façon de faire vous semble louche ou pas usuelle, je me laisse guider sur vos propositions.
J'ai pas trouvé mieux comme compromis "respect de comment Ember Data fonctionne" / ce que l'on veut atteindre 😅

Reply:
La liste a ete rajouter via l'utilisateur. Ca ne viens pas du back, on peut simplifier
```suggestion
const relatedBadges = complementaryCertification.complementaryCertificationBadges ?? [];
```